### PR TITLE
Move project template files in internal folder - close #308

### LINKED ISSUE #308: Move runtime files in project template
When the CLI updates a project, the following items should be easily replaced, so moving them in an internal folder will make this easier.

* `metadataGenerator`
* `NativeScript`
* `NativeScript.framework`
* `TNSDebugging.h`

### PR DESCRIPTION
**Pull request by jasssonpet**
New package structure:

```
tns-ios.tgz
├── WebInspectorUI
│   ├── NativeScript Inspector.zip
│   └── Safari
└── framework
    ├── __PROJECT_NAME__
    │   ├── __PROJECT_NAME__-Info.plist
    │   ├── __PROJECT_NAME__-Prefix.pch
    │   ├── build.xcconfig
    │   └── en.lproj
    │       ├── InfoPlist.strings
    │       └── LaunchScreen.xib
    ├── __PROJECT_NAME__.xcodeproj
    │   ├── project.pbxproj
    └── internal
        ├── NativeScript
        │   ├── Frameworks
        │   │   └── NativeScript.framework
        │   │       ├── Headers
        │   │       │   ├── NativeScript.h
        │   │       │   ├── TNSRuntime+Inspector.h
        │   │       │   └── TNSRuntime.h
        │   │       ├── Info.plist
        │   │       ├── NativeScript
        │   │       └── _CodeSignature
        │   │           └── CodeResources
        │   ├── include
        │   │   ├── NativeScript.h
        │   │   ├── TNSRuntime+Inspector.h
        │   │   └── TNSRuntime.h
        │   └── lib
        │       └── libNativeScript.a
        ├── TNSDebugging.h
        ├── main.m
        ├── metadata-generator
        │   ├── bin
        │   │   ├── metadata-generation-build-step
        │   │   └── objc-metadata-generator
        │   └── lib
        │       └── clang
        │           └── 3.6.0
        │               └── include
        ├── nativescript-build.xcconfig
        ├── nativescript-post-build
        ├── nativescript-pre-build
        ├── nativescript-pre-link
        └── strip-dynamic-framework-architectures.sh
```

Related to https://github.com/NativeScript/nativescript-cli/pull/951.

### CODE REVIEW

File: build/project-template/__PROJECT_NAME__.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,17 +7,9 @@
...
-		858B842A18CA22B800AB12DE /* UIKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 858B833518CA111C00AB12DE /* UIKit.framework */; settings = {ATTRIBUTES = (Weak, ); }; };
 		858B842D18CA22B800AB12DE /* InfoPlist.strings in Resources */ = {isa = PBXBuildFile; fileRef = 858B833A18CA111C00AB12DE /* InfoPlist.strings */; };
-		CD1343101AF8E03800D3B65B /* libNativeScript.a in Frameworks */ = {isa = PBXBuildFile; fileRef = CD13430F1AF8E03800D3B65B /* libNativeScript.a */; };
 		CD45EE7C18DC2D5800FB50C0 /* app in Resources */ = {isa = PBXBuildFile; fileRef = CD45EE7A18DC2D5800FB50C0 /* app */; };
-		F9161D911993C00300F9E5FD /* libc++.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = F9161D901993C00300F9E5FD /* libc++.dylib */; };
-		F9161D931993C00B00F9E5FD /* libz.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = F9161D921993C00B00F9E5FD /* libz.dylib */; };
-		F9161D951993C01600F9E5FD /* libicucore.dylib in Frameworks */ = {isa = PBXBuildFile; fileRef = F9161D941993C01600F9E5FD /* libicucore.dylib */; };
-		F9B8C6A41ACEDEEB00547416 /* LaunchScreen.xib in Resources */ = {isa = PBXBuildFile; fileRef = F9B8C6A21ACEDEEB00547416 /* LaunchScreen.xib */; };
```
Comment (jasssonpet) [inline build/project-template/__PROJECT_NAME__.xcodeproj/project.pbxproj]: This line got replaced by Xcode 7 every time. The file is referenced elsewhere in the project.

**Comment by ivanbuhov** [comment]
:+1:

**Comment by fealebenpae** [comment]
I wholeheartedly approve, although I want to make a small suggestion:
```
    NativeScript
    ├── Frameworks
    │   └── NativeScript.framework
    │       ├── Headers
    │       │   ├── NativeScript.h
    │       │   ├── TNSRuntime+Inspector.h
    │       │   └── TNSRuntime.h
    │       ├── Info.plist
    │       ├── NativeScript
    │       └── _CodeSignature
    │           └── CodeResources
    ├── include
    │   ├── NativeScript.h
    │   ├── TNSRuntime+Inspector.h
    │   └── TNSRuntime.h
    └── lib
        └── libNativeScript.a
```

**Comment by jasssonpet** [comment]
@ I've moved the `NativeScript.framework` folder inside `NativeScript/Frameworks`.

**Comment by fealebenpae** [comment]
Alright, I suggest we merge this. Could you please open an issue with the CLI so that they know that the platform update procedure for 1.4 needs to be tweaked again, and let the QAs know that the layout of the npm package has changed?

**Comment by jasssonpet** [comment]
There is already a PR in the CLI with the updated project structure: https://github.com/NativeScript/nativescript-cli/pull/951

**Comment by fealebenpae** [comment]
Well, good work, you've earned a muffin :Д

**Pull request merged**